llvm/test/Other/dump-before-after.ll

```diff
@@ -0,0 +1,56 @@
+; RUN: rm -rf %t/logs
+
+; Basic dump before and after a single module pass
+
+; RUN: opt %s -disable-output -passes='no-op-module' -ir-dump-directory %t/logs -print-after=no-op-module -print-before=no-op-module
```

Comment:
relying on specific `find` flags can be not very portable. the following is simpler and probably more portable:
```
 ; RUN: opt %s -disable-output -passes='no-op-module' -ir-dump-directory %t/logs -print-after=no-op-module -print-before=no-op-module
-; RUN: find %t/logs -type f -printf '%P\n' | sort -n | FileCheck %s --check-prefix=SINGLE-PASS
-; SINGLE-PASS: {{[0-9]+}}-[[MODULE_NAME_HASH:[a-z0-9]+]]-module-NoOpModulePass-after.ll
-; SINGLE-PASS: {{[0-9]+}}-[[MODULE_NAME_HASH]]-module-NoOpModulePass-before.ll
+; RUN: ls %t/logs | FileCheck %s --check-prefix=SINGLE-PASS
+; RUN: ls %t/logs | count 2
+; SINGLE-PASS-DAG: {{[0-9]+}}-[[MODULE_NAME_HASH:[a-z0-9]+]]-module-NoOpModulePass-after.ll
+; SINGLE-PASS-DAG: {{[0-9]+}}-[[MODULE_NAME_HASH]]-module-NoOpModulePass-before.ll
 ; RUN: rm -rf %t/logs
```

Reply:
As mentioned in another thread: https://github.com/llvm/llvm-project/pull/66412#discussion_r1326511891, `DAG` checks don't enforce what I want.

I might be able to use `ls` instead of `find` though, let me try.

Reply:
`ls` worked just fine.

Reply:
the actual numbers should be consistent since they depend solely on the passes that run?

Reply:
Yes, but I'm not sure how stable the passes that run are. Even though I only ask to run various no-op passes, a couple more run implicitly. If that changes, and we assert the numbers themselves, the test will break:

```
0-d2b22a3c72db5719-module-NoOpModulePass-after.ll
1-d2b22a3c72db5719-scc-815922cdb8ef56e8-NoOpCGSCCPass-after.ll
2-d2b22a3c72db5719-scc-5525c0ec6c125513-NoOpCGSCCPass-after.ll
3-d2b22a3c72db5719-function-dcb27518fed9d577-NoOpFunctionPass-after.ll
4-d2b22a3c72db5719-function-003934191339461a-NoOpFunctionPass-after.ll
5-d2b22a3c72db5719-function-dcb27518fed9d577-LoopSimplifyPass-after.ll
6-d2b22a3c72db5719-function-dcb27518fed9d577-LCSSAPass-after.ll
7-d2b22a3c72db5719-function-003934191339461a-LoopSimplifyPass-after.ll
8-d2b22a3c72db5719-function-003934191339461a-LCSSAPass-after.ll
9-d2b22a3c72db5719-loop-f8786d034c39b728-NoOpLoopPass-after.ll
10-d2b22a3c72db5719-module-NoOpModulePass-after.ll
11-d2b22a3c72db5719-module-VerifierPass-after.ll
```

Reply:
all of those passes are expected to run and should be consistent, although I'm removing the last two in https://github.com/llvm/llvm-project/pull/67730

Reply:
Removed `find` and `sort` usage, check the actual numbers now.